cost/aws/object_storage_optimization/aws_object_storage_optimization.pt

```diff
@@ -149,7 +155,62 @@ datasource "ds_aws_buckets_with_region" do
 end
 
 datasource "ds_aws_sanitized_buckets" do
-  run_script $js_parse_buckets, $ds_aws_buckets_with_region
+  run_script $js_parse_buckets, $ds_aws_buckets_with_region, $param_bucket_list
+end
+
+script "js_parse_buckets", type: "javascript" do
+  parameters "buckets", "param_bucket_list"
+  result "results"
+  code <<-EOS
+    results = []
+    if (param_bucket_list.length > 0){
```

Comment:
There's a lot of duplicate code here between the "if" section and the "else" section. Can this be consolidated in such a way that the if-else mostly just handles the difference between the two?

One method would be to use underscore.js to filter the bucket list via _.filter (using an if-else statement to either create a new filtered list or just set the list equal to the ds_aws_buckets_with_region parameter), and then do all the other work after the if-else statement:

https://underscorejs.org/#filter

Reply:
Hi Shawn, Thank you for the review, changes are addressed.